### PR TITLE
Upgraded AEM API for AEM 6.2

### PR DESCRIPTION
**Pull request by ethomasjoseph**
Creating reference to the AEM 6.2 API

### CODE REVIEW

**Comment by jbaix** [comment]
It whould be great to have a simple mechanism to switch between different versions of AEM API using this archetype, also considering obfuscated and unobfuscated apis.

**Comment by rombert** [comment]
Right, one of the reasons for not merging is that I'd push everyone to 6.2, which is not the case yet - I think we should still support 6.0.

@gabrielwalt - do we want to support all versions which are in core support as per https://www.adobe.com/support/products/enterprise/eol/eol_matrix.html#cq or just the most recent versions?

**Comment by joerghoh** [comment]
@rombert I would drop support to create projects for old AEM versions; older versions of this archetype can be used for it.

**Comment by bpauli** [comment]
We should still support older versions of AEM with new features and fixes of the archetype.

Maybe we can add some kind of semantic versioning to the archetype to indicate which AEM version is supported.

Proposal for the next release (minor version is mapped to AEM minor version). Each minor version is maintained in a separate branch. 
For example:
11.0.0 -> AEM 6.0
11.1.0 -> AEM 6.1
11.2.0 -> AEM 6.2
11.3.0 -> upcoming 6.3 release

bugfixes go into next patch version  (e.g. 11.0.1). For new features the major version will be increased. 
WDYT?

**Comment by joerghoh** [comment]
I don't think that many projects for AEM 6.0 are still being created.

**Comment by rombert** [comment]
I tend to agree with @joerghoh - projects for older versions can use an older archetype version. Also, maintaining more versions in parallel means more work, and I don't see that happening :-)

**Comment by kwin** [comment]
I would also recommend to only support the latest AEM release with the latest release of the archetype.

File: src/main/archetype/pom.xml [hunk 1]
```diff
@@ -512,8 +512,9 @@
             <!-- Apache Sling Dependencies -->
             <dependency>
                 <groupId>com.adobe.aem</groupId>
-                <artifactId>aem-api</artifactId>
-                <version>6.0.0.1</version>
+                <artifactId>uber-jar</artifactId>
+                <version>6.2.0</version>
+                <classifier>obfuscated-apis</classifier>
```
Comment (bpauli) [inline src/main/archetype/pom.xml]: Can you please change the classifier to just apis [0]. IMO as the uber-jar is now published on the public git repository repo.adobe.com there is now reason to only include the obfuscated one.

[0] https://docs.adobe.com/docs/en/aem/6-2/develop/dev-tools/ht-projects-maven.html#What%20is%20the%20UberJar?

File: src/main/archetype/it.tests/pom.xml [hunk 2]
```diff
@@ -108,7 +108,8 @@
         
         <dependency>
             <groupId>com.adobe.aem</groupId>
-            <artifactId>aem-api</artifactId>
+            <artifactId>uber-jar</artifactId>
+            <classifier>obfuscated-apis</classifier>
```
Comment (bpauli) [inline src/main/archetype/it.tests/pom.xml]: see comment in parent pom

File: src/main/archetype/core/pom.xml [hunk 3]
```diff
@@ -91,7 +91,8 @@
         </dependency>
         <dependency>
             <groupId>com.adobe.aem</groupId>
-            <artifactId>aem-api</artifactId>
+            <artifactId>uber-jar</artifactId>
+            <classifier>obfuscated-apis</classifier>
```
Comment (bpauli) [inline src/main/archetype/core/pom.xml]: see comment in parent pom

File: src/main/archetype/ui.apps/pom.xml [hunk 4]
```diff
@@ -143,7 +143,8 @@
 
         <dependency>
             <groupId>com.adobe.aem</groupId>
-            <artifactId>aem-api</artifactId>
+            <artifactId>uber-jar</artifactId>
+            <classifier>obfuscated-apis</classifier>
```
Comment (bpauli) [inline src/main/archetype/ui.apps/pom.xml]: see comment in parent pom

File: src/main/archetype/ui.content/pom.xml [hunk 5]
```diff
@@ -143,7 +143,8 @@
 
         <dependency>
             <groupId>com.adobe.aem</groupId>
-            <artifactId>aem-api</artifactId>
+            <artifactId>uber-jar</artifactId>
+            <classifier>obfuscated-apis</classifier>
```
Comment (bpauli) [inline src/main/archetype/ui.content/pom.xml]: see comment in parent pom

**Comment by rombert** [comment]
Merged in 94ea7fd6db9f2d86f7b2d4df683177cee79a59e6, thanks for the contribution! Manually applied @bpauli's suggestions regarding using the _apis_ classifier.

**Pull request closed without merging**